go/host/client_api_eth.go

```diff
@@ -45,17 +47,24 @@ func (api *EthereumAPI) GetBalance(_ context.Context, encryptedParams common.Enc
 
 // GetBlockByNumber returns the rollup with the given height as a block. No transactions are included.
 func (api *EthereumAPI) GetBlockByNumber(_ context.Context, number rpc.BlockNumber, _ bool) (map[string]interface{}, error) {
-	extRollup, err := api.host.EnclaveClient.GetRollupByHeight(number.Int64())
-	if extRollup == nil {
-		return nil, err
+	rollupHash := api.host.nodeDB.GetRollupHash(big.NewInt(int64(number)))
+	if rollupHash == nil {
+		return nil, nil //nolint:nilnil
+	}
+	rollupHeaderWithHashes := api.host.nodeDB.GetRollupHeader(*rollupHash)
+	if rollupHeaderWithHashes == nil {
+		return nil, fmt.Errorf("could not retrieve header for stored rollup with number %d and hash %s", number, rollupHash)
 	}
-	return extRollupToBlock(extRollup), err
+	return headerWithHashesToBlock(rollupHeaderWithHashes), nil
 }
 
 // GetBlockByHash returns the rollup with the given hash as a block. No transactions are included.
 func (api *EthereumAPI) GetBlockByHash(_ context.Context, hash gethcommon.Hash, _ bool) (map[string]interface{}, error) {
-	extRollup, err := api.host.EnclaveClient.GetRollup(hash)
```

Comment:
Tangential to the changes here, but saw a good opportunity to avoid a hop to the enclave.